lightning/src/routing/router.rs

```diff
@@ -1181,6 +1181,11 @@ pub struct PrivateHopCandidate<'a> {
 /// A [`CandidateRouteHop::Blinded`] entry.
 #[derive(Clone, Debug)]
 pub struct BlindedPathCandidate<'a> {
+	/// The node id of the introduction node, resolved from either the [`NetworkGraph`] or first
+	/// hops.
+	///
+	/// This is not exported to bindings users as lifetimes are not expressable in most languages.
+	pub source_node_id: &'a NodeId,
```

Comment:
Hmm... I think we want to avoid putting 33-bytes in this struct, thus why we try to use references. (cc: @TheBlueMatt)

Regarding the clones, fortunately the `Vec` only stores references, so we aren't cloning `ChannelDetails` just allocating a new `Vec` holding their references. If the `find_route` took `RouteParameters` by value, we could do a first pass to resolve `IntroductionNode::DirectedShortChannelId` to `IntroductionNode::NodeId`. So to do that with the current interface, we'd need to clone `RouteParameters`. Not sure if that's any better. 🙁